src/Http/Http/src/Features/HttpRequestIdentifierFeature.cs

```diff
@@ -34,14 +34,14 @@ public string TraceIdentifier
             }
         }
 
-        private static unsafe string GenerateRequestId(long id)
+        private static string GenerateRequestId(long id)
         {
             // The following routine is ~310% faster than calling long.ToString() on x64
             // and ~600% faster than calling long.ToString() on x86 in tight loops of 1 million+ iterations
             // See: https://github.com/aspnet/Hosting/pull/385
 
             // stackalloc to allocate array on stack rather than heap
-            char* charBuffer = stackalloc char[13];
+            Span<char> charBuffer = stackalloc char[13];
 
             charBuffer[0] = _encode32Chars[(int)(id >> 60) & 31];
```

Comment:
While you are on it, you could remove the casts to `int`. 
Without the casts the JIT produces better code (not containing `movsxd`s) -- for 64bit systems perf rises.
For 32bit systems additional code may be generated, resulting in worse perf.

I don't know how perf-critical these codes are, so this comment may be ignored.

Reply:
> as the indexer expect an int, not a long.

Really? See [Spec for Array element access](https://github.com/dotnet/csharplang/blob/master/spec/arrays.md#array-element-access) :wink:

Reply:
Ah sorry. You are in `string`, I'm on `char[] `, see the other comment from me.

Reply:
done